### PR TITLE
Dev 18195

### PR DESCRIPTION
**Pull request by khelmrich**
Main codebase without test. Next PR will be with tests

### CODE REVIEW

**Review by jacekgadomski — CHANGES_REQUESTED** [review]
nice, some minor things to fix

File: src/Import/Domain/DuckDb/FileConverter.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Productsup\BinCdeAmazonS3Parquet\Import\Domain\DuckDb;
+
+interface FileConverter
+{
```
[the remark this answers is not in the document]
Comment (khelmrich) [reply src/Import/Domain/DuckDb/FileConverter.php:8]: moved to application layer and renamed

File: src/Import/Infrastructure/Transporter/Exception/ConnectionException.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Productsup\BinCdeAmazonS3Parquet\Import\Infrastructure\Transporter\Exception;
+
+use Productsup\DK\Connector\Exception\Core\ClientLevelException;
+use Throwable;
+
+final class ConnectionException extends ClientLevelException
```
[the remark this answers is not in the document]
Comment (khelmrich) [reply src/Import/Infrastructure/Transporter/Exception/ConnectionException.php:10]: it was not used I deleted this class

File: src/Import/Infrastructure/Transporter/Exception/ConnectionException.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public static function dueToPrevious(Throwable $previous): self
+    {
+        return new self(
+            message: $previous->getMessage(),
```
[the remark this answers is not in the document]
Comment (khelmrich) [reply src/Import/Infrastructure/Transporter/Exception/ConnectionException.php:15]: class not needed, has been deleted

File: Dockerfile [hunk 4]
```diff
@@ -5,6 +5,13 @@ COPY config/ ./config
 COPY bin/ ./bin
 COPY .env composer.json composer.lock symfony.lock ./
 
+RUN apt-get update && apt-get install -y jq
+
+RUN wget https://github.com/duckdb/duckdb/releases/download/v1.0.0/duckdb_cli-linux-amd64.zip \
```
[the remark this answers is not in the document]
Comment (khelmrich) [reply Dockerfile:10]: https://productsup.atlassian.net/browse/DEV-18652

I have created ticket for this

File: config/services.yaml [hunk 5]
```diff
@@ -1,10 +1,18 @@
...
 services:
     _defaults:
         autowire: true
         autoconfigure: true
+        bind:
+            $accessKeyId: '%env(ACCESS_KEY_ID)%'
+            $secretAccessKey: '%env(SECRET_ACCESS_KEY)%'
+            $bucket: '%env(BUCKET)%'
+            $region: '%env(REGION)%'
+            $tempFilename: '%temp_filename%'
```
[the remark this answers is not in the document]
Comment (khelmrich) [reply config/services.yaml:10]: I am using it in two places. I want to make sure that it is the same for both classes. One is downloading second converting it.

File: src/Import/Infrastructure/Transporter/Exception/DownloadFailed.php [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Productsup\BinCdeAmazonS3Parquet\Import\Infrastructure\Transporter\Exception;
+
+use Productsup\DK\Connector\Exception\Core\EngineeringLevelException;
+use Throwable;
+
+final class DownloadFailed extends EngineeringLevelException
```
[the remark this answers is not in the document]
Comment (khelmrich) [reply src/Import/Infrastructure/Transporter/Exception/DownloadFailed.php:10]: not exist I will add it to the framework

File: .env [hunk 7]
```diff
@@ -1 +1,6 @@
 APP_ENV=prod
+FILE_NAME=
+AWS_ACCESS_KEY_ID=
+AWS_SECRET_ACCESS_KEY=
```
Comment (jacekgadomski) [inline .env:4]: rememeber to add this as a secret in the connector config

File: config/services.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (jacekgadomski) [reply config/services.yaml:10]: ah ok i thought that it is configurable

File: src/Import/Infrastructure/Transporter/Exception/DownloadFailed.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (jacekgadomski) [reply src/Import/Infrastructure/Transporter/Exception/DownloadFailed.php:10]: when? it would be good to add it now and do not have it here

**Review by jacekgadomski — APPROVED** [review]

File: .env (hunk 7 above)
Comment (khelmrich) [reply .env:4]: I will :)

**Pull request merged**